KHLY2700/KHLY2700-IOC-01App/Db/keithley2700_channels.template

```diff
@@ -42,11 +46,87 @@ record(ao, "$(P)$(CHANNEL):TIME")
 {
 	field(DESC, "Latest device reading timestamp")
 	field(EGU, "")
+	field(PREC, "7")
+	info(INTEREST, "HIGH")
+	info(ARCHIVE, "VAL")
+	field(FLNK, "$(P)$(CHANNEL):TEMP:RAW")
+}
+
+record(ao, "$(P)$(CHANNEL):TIME:PREV")
+{
+	field(DESC, "Previous timestamp value")
+	field(EGU, "")
+	field(PREC, "5")
+}
+
+record(cvt, "$(P)$(CHANNEL):TEMP:RAW")
+{
+  field(DESC, "Calculate temperature from calib file")
+  field(SCAN, "Passive")
+  field(PREC, "5")
+  field(EGU, "K")
+  field(INPY, "$(P)$(CHANNEL):READ")
+  field(METH, "1D TABLE INVERTED")
+  field(DRVL, "$(DRVLO)")
+  field(DRVH, "$(DRVHI)")
+  field(SPEC, "$(CALIB_FILE)")
+  field(TDIR, "$(CALIB_DIR)")
+  field(BDIR, "$(CALIB_BASE_DIR)")
+  info(INTEREST, "LOW")
+  field(FLNK, "$(P)$(CHANNEL):TEMP:CHECK")
+}
+
+record(calcout, "$(P)$(CHANNEL):TEMP:CHECK")
+{
+	field(DESC, "Check temp is within range")
+	field(OOPT, "Every Time")
+	field(INPA, "$(P)$(CHANNEL):TEMP:RAW")
+	field(CALC, "((A>=0)&&(A<320))?A:0")
```

Comment:
It would be good to leave a comment as to why these specific values are used (I assume device limits). If outside of these limits you might want to alarm, otherwise if maybe the device is on fire it might read 0.

Reply:
Added alarm, test, and comment